libs/headless-lfp/src/language-font-picker.ts

```diff
@@ -38,7 +40,29 @@ export function useLanguageFontPicker(
     [extraFonts, lff, maxFontCount]
   );
 
-  return { fetchFonts, fonts };
+  /** Toggle whether the specified font is selected.
+   * If `options.multiselect` is `true`, leave all other fonts alone.
+   * If `options.multiselect` is `false`, set `selected: false` on all other fonts.
+   * If specified font is in the fonts list, return the resulting `selected` value of that font.
+   * If specified font is not in the fonts list, return `undefined`.  */
+  const toggleSelectFont = useCallback(
+    (fontName: string) => {
+      let selected: boolean | undefined;
+      setFonts((prev) =>
+        prev.map((font) => {
+          if (font.name === fontName) {
+            selected = !font.selected;
+            return { ...font, selected };
+          }
+          return multiselect ? font : { ...font, selected: false };
```

Comment:
I was initially confused that this code wasn't doing what it says in L45 because it's doing the opposite of the Return Early Pattern. I would change it to follow the pattern, i.e.:
```typescript
          if (font.name !== fontName) {
            return multiselect ? font : { ...font, selected: false };
          }

          selected = !font.selected;
          return { ...font, selected };
```